src/provider/provider_devdax_memory_internal.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#ifndef UMF_DEVDAX_MEMORY_PROVIDER_INTERNAL_H
+#define UMF_DEVDAX_MEMORY_PROVIDER_INTERNAL_H
+
+#include <umf/providers/provider_os_memory.h>
```

Comment:
We should not have dependencies between providers. If we need share code, we should move it to utils.